addons/website/static/src/snippets/s_image_gallery/options.js

```diff
@@ -317,13 +318,32 @@ options.registry.gallery = options.Class.extend({
                     imgs.push(data.$image[0]);
                     break;
             }
+            position = _.indexOf(imgs, data.$image[0]);
             _.each(imgs, function (img, index) {
                 // Note: there might be more efficient ways to do that but it is
                 // more simple this way and allows compatibility with 10.0 where
                 // indexes were not the same as positions.
                 $(img).attr('data-index', index);
             });
-            this.mode('reset', this.getMode());
+            const currentMode = this.getMode();
+            this.mode('reset', currentMode);
+            if (currentMode === 'slideshow') {
+                const $carousel = this.$target.find('.carousel');
+                $carousel.removeClass('slide');
+                $carousel.carousel(position);
+                this.$target.find('.carousel-indicators li').removeClass('active');
+                this.$target.find('.carousel-indicators li[data-slide-to="' + position + '"]').addClass('active');
+                this.trigger_up('activate_snippet', {
+                    $snippet: this.$target.find('.carousel-item.active img'),
+                    ifInactiveOptions: true,
+                });
```

Comment:
why does

```javascript
this.trigger_up('activate_snippet', {
    $snippet: data.$image,
    ifInactiveOptions: true,
});
```

not work in this case? The active image should be data.$image after using `carousel(position)` right? So you could just remove this and execute what's the the else in both cases.

Reply:
Eh eh, I asked the exact same question to him a while ago ;) If I remember well, I investigated myself and found that the call to `this.mode` above rebuilds everything and so `data.$image` references the image before rebuilding. Should be refactored if possible, if too difficult, this can be ok.

Reply:
I will leave like this ;)